### ISSUE TITLE
Custom code style

### ISSUE THREAD

**Issue by krlmlr** [igraph/rigraph#1340]
For graph literals.

https://github.com/igraph/rigraph/pull/673 contains the logic that was used in a one-off run. It would be nice to have the code styled on a regular basis, e.g., in a pre-commit hook.

**Comment by maelle** [comment]
In the PR, there are two scripts defining the same function `normalize_igraph_arrows()`.

**Comment by maelle** [comment]
I'd suggest:

- creating a package holding the custom style (like https://github.com/ropensci-review-tools/spaceout)
- referencing it in the contributing guide. Regular contributors would install and run it locally.
- creating and tweaking a PR command workflow https://github.com/r-lib/actions/blob/473c68190595b311a74f208fba61a8d8c0d4c247/examples/pr-commands.yaml#L51

Does that sound good @krlmlr?

**Comment by krlmlr** [comment]
Can that be a function in the igraph package too?

**Comment by maelle** [comment]
I suppose it could, but I find it cleaner this way. https://github.com/igraph/igraph.r2cdocs is a separate package for instance.

If it is a function in the igraph package, could we export it? Styling graphs might be relevant for igraph users too.

**Comment by krlmlr** [comment]
Yeah, let's start with a separate package. Merging is easier than tearing apart.

We have infrastructure to run styler on every PR, need to understand how to make it work with that new package or style guide. No PR command workflow needed.

**Comment by maelle** [comment]
@krlmlr are all these snaps correct? https://github.com/igraph/igraph.style/blob/main/tests/testthat/_snaps/igraph-style.md

In the PR you had linked to, I think all versions of `normalize_igraph_arrows()` were the same but they contained leftover `browser()`.

**Comment by krlmlr** [comment]
- Needs a space before the RHS: `A +-+ B` etc.
- Also: `A -+ B:C`

**Comment by maelle** [comment]
Right, so does it mean you had used other code in your PR? I'm confused.

**Comment by maelle** [comment]
oh maybe the functions aren't the same (I had tried to compare them with version control, maybe I pasted the wrong one :melting_face: )

**Comment by maelle** [comment]
RHS problem fixed but did the styling of `styler::style_text("A - +B:C", style = igraph_style)` ever work?

**Comment by krlmlr** [comment]
Might not have worked ever, or styler changed in the meantime.

**Comment by maelle** [comment]
Do you still remember enough about the code to fix {igraph.style} yourself or should I try?

**Comment by krlmlr** [comment]
I don't have the capacity to deal with that at the moment.

**Comment by krlmlr** [comment]
Seems to work now. Next steps:

- [ ] Run locally on the package, PR
- [ ] Decide on a level of quality (strict, scope)
- [ ] Configure GHA to run automatically

I can help with the third step.

**Comment by maelle** [comment]
I ran it in https://github.com/igraph/rigraph/pull/1696 and it went really well

**Comment by schochastics** [comment]
obsolete due to air
